src/Storages/Statistics/StatisticsCountMinSketch.cpp

```diff
@@ -0,0 +1,111 @@
+#include <DataTypes/DataTypeLowCardinality.h>
+#include <DataTypes/DataTypeNullable.h>
+#include <IO/ReadHelpers.h>
+#include <IO/WriteHelpers.h>
+#include <Storages/Statistics/StatisticsCountMinSketch.h>
+#include <Interpreters/convertFieldToType.h>
+#include <DataTypes/FieldToDataType.h>
+
+#if USE_DATASKETCHES
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+extern const int ILLEGAL_STATISTICS;
+extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+}
+
+/// Constants chosen based on rolling dices, which provides an error tolerance of 0.1% (ε = 0.001) and a confidence level of 99.9% (δ = 0.001).
+/// And sketch the size is 152kb.
+static constexpr auto num_hashes = 7uz;
+static constexpr auto num_buckets = 2718uz;
+
+StatisticsCountMinSketch::StatisticsCountMinSketch(const SingleStatisticsDescription & stat_, DataTypePtr data_type_)
+    : IStatistics(stat_)
+    , sketch(num_hashes, num_buckets)
+    , data_type(data_type_)
+{
+}
+
+Float64 StatisticsCountMinSketch::estimateEqual(const Field & val) const
+{
+    if (data_type->isValueRepresentedByNumber())
+    {
+        /// 'val' maybe number or string, method 'convertFieldToType' will
+        ///     1. convert string to number, date, datetime, IPv4, Decimal etc
+        ///     2. return null if val larger than the range of data_type
+        auto val_converted = convertFieldToType(val, *data_type);
+        if (val_converted.isNull())
+            return 0;
+
+        /// We will get the proper data type of val_converted, for example, UInt8 for 1, UInt16 for 257.
+        auto data_type_converted = applyVisitor(FieldToDataType<LeastSupertypeOnError::Null>(), val_converted);
+        DataTypes data_types = {data_type, data_type_converted};
+        auto super_type = tryGetLeastSupertype(data_types);
```

Comment:
for l46 - l49, if we compare `a = 1.0` or `a = cast(0 as Int64)` when a is Int32. we are able to estimate it.

Reply:
If type of a is Int32:
1. For `a = 1.0` `val_converted` is `Int64,  `data_type_converted` is `Int8`, `super_type` is `Int32`
2. For  `a = cast(0 as Int64)` `val_converted` is `Int64`, `data_type_converted` is `Int8`, `super_type` is `Int32`
3. For `a = 1.1``val_converted` is `Null` for it can not be converted to `Int32`
4. For `a = 257``val_converted` is `Int64`, `data_type_converted` is `Int16`, `super_type` is `Int32`
5. For `c = max_value_int64` `val_converted` is `Int64`, `data_type_converted` is `Int64`, `super_type` is `Int64`
6. For `c = '1'` `val_converted` is `Int64`, `data_type_converted` is `Int8`, `super_type` is `Int32`